### DISCUSSION TITLE
Current Gatsby Support?

### DISCUSSION THREAD

**Discussion by xaviemirmon** [tinacms/tinacms#2346] [General]
Hey Tinacms Team!

It's been a little while since I worked with TinaCMS and Gatsby.  I'm an avid fan of the product and have followed your progress since the early days.  Looking around I now see that Gatsby has fallen out of favor and now you are focussing Next.js. 

> Note: Building with Tina and Gatsby currently requires that you run a development server for content-editing. We recommend using Next.js 5 for a solution with less friction.

I am curious about a few things around this decision and how I might go about helping keep the Gatsby support alive.  The   

What specific technicalities led you to land on Next.js over Gatsby as your preferred React framework? Is that documented somewhere?
Has Gatsby + TinaCMS support officially ended and all Gatsby packages are unmaintained and unsupported?  I can see that the packages no longer exist in the main TinaCMS repo but are still on NPM. 
If the answer above is yes, how would you suggest going about reviving this support? Should I fork the existing packages as a new project or contribute back to the main TinaCMS repo via PRs.

I'm starting work on an exciting new project with a Drupal backend and Gatsby backend and would love to understand the best to bring TinaCMS into this fold.  

Any thoughts on this would be much appreciated!

**Comment by scottgallant** [comment]
Hi Xavier, nice to see you again! 

Unfortunately we had to stop supporting Gatsby due to Technical limitations.  In the future, we intend to support Gatsby only with a traditional CMS UI and not our visual editing UI (the UI will support these two modes in early 2022. See prototype Gif below where I toggle between the two by clicking the full-screen button).

![NnN0XLm8r1](https://user-images.githubusercontent.com/776019/146267260-67ddaff3-4a9c-4ba7-a0e0-619e52c8adea.gif)

Others from our team can speak to this better than I can but put simply, Tina's visual editing features work best with Reactive frameworks (React/Vue/Svelte-based) that are data agnostic (can fetch data from anywhere without strong opinions).  For frameworks that don't meet those criteria (i.e. Gatsby), Tina will only work in Standard mode (no visual editing) which will be coming early next year. 

Are you open to choosing a new Framework?

**Reply by alexhendel** [reply]
Hi Scott,

On the topic of currently supported frameworks, I was wondering what the current state of support for `create-react-app` is? 

I did use an earlier guide for implementing TinaCMS (0.37) in a CRA app. I was trying to update this app to the recent Tina version but failed to do so. There seem to be too many breaking changes. And I couldn't find any current guides in the Docs. Is CRA something currently working/supported with recent Tina versions?

Are there any guides on how to migrate from 0.3x to a recent version?

**Reply by jamespohalloran** [reply]
Hey Alex,

`create-react-app` won't work with the latest version of tinacms, because currently with Tina, your production content needs to be loaded statically. We have "Read-only tokens" on our roadmap, which would open things up so that you could fetch from the Tina GraphQL API client-side. We hope to share more on this in the new year, as we develop our public roadmap.

**Reply by alexhendel** [reply]
Thank you, James. I'll wait and see what's to come. 👍

**Comment by xaviemirmon** [comment]
Hi @scottgallant and @jamespohalloran,

It's great to be back! So currently the plan is to use `react-tinacms-inline` with a Drupal 9 backend that builds out to GatsbyJS.  I've currently got a bit of a PoC together where I am using Gatsby V4's new SSR functionality to provide data to Tina and it's working quite well.

Drupal is data storage backend, Tina is the editing experience and Gatsby the frontend. I've heard on another ticket from @jeffsee55 that `react-inline-forms` is also going to be dropped from your support/roadmap which potentially opens up this discussion a bit wider.  I completely understand that the Tina roadmap needs align the greater goals for Forestry and is probably forming where you focus your efforts.   For the parts of the project that Forestry cannot commit further resources to, how best would you suggest that the community keeps them alive?  

Should it be forked to new project? Or, is there a way to commit these contributions back to the project and revive some of the depecrated bits you've paused/stopped working on for now.

@scottgallant as an aside I'd love to arrange show you what we've been working on.  We had a call with the team about it probably about 3 years ago now and I'm finally in a position to make it happen!

**Reply by jeffsee55** [reply]
@xaviemirmon I wouldn't mind seeing this, you can reach out to me directly at jeff@forestry.io or join [our Discord](https://discord.gg/yGQM4G2R) where we can arrange a chat

**Comment by panzacoder** [comment]
Just dropping in to say I'm following along. I totally understand the limitations with Gatsby and Tina, and I met with the team over a year ago to talk about my use-case and the story was the same... "Have you considered Next?"

Gatsby continues to prove valuable to me and my clients so opportunities to improve the CMS experience are always interesting to me as well. When full SSR support came about I thought of Tina immediately. If you start a fork of this repo @xaviemirmon I'd be interested to check it out.

**Reply by scottgallant** [reply]
👋 hey, we're going to start exploring other frameworks very soon. I just wrote about that today [here](https://tina.io/blog/tinacms-2022/). I didn't mention Gatsby specifically because we haven't tried Gatsby with SSR and Tina yet, but theoretically, it should work! 

Are you starting a new project that you're hoping to test Tina on now, or are you just curious? 

Thanks for your patience!
